sdk/inc/azure/core/az_span.h

```diff
@@ -76,17 +76,24 @@ AZ_NODISCARD az_span az_span_create(uint8_t* ptr, int32_t size);
 #endif // AZ_NO_PRECONDITION_CHECKING
 
 /**
- * @brief An empty #az_span.
+ * @brief An empty #az_span literal.
  *
  * @remark There is no guarantee that the pointer backing this span will be `NULL` and the caller
  * shouldn't rely on it. However, the size will be 0.
  */
-#define AZ_SPAN_EMPTY                      \
-  (az_span)                                \
+#define AZ_SPAN_LITERAL_EMPTY              \
```

Comment:
Good callout. I believe we needed this when we had nested literal structs but we have since changed some fields to no longer need this. Let me double check in this.

Reply:
I stand corrected. This is needed for the same reason we have `AZ_SPAN_LITERAL_FROM_STR()`

![image](https://user-images.githubusercontent.com/36710865/199521177-fc1fcddc-73fb-42ee-af29-96c790ad1cb3.png)

It will complain that the value is not constant since it has a cast to an `az_span`.